zed-extensions GitHub action: `bump_extension_version` should only bump the extension version, and ignore third party `[dependencies]`

### Reproduction steps

1. When publishing an extension in `zed-extensions`, the Zed-provided GitHub action (I think it is based on https://github.com/zed-industries/zed/blob/main/.github/workflows/extension_bump.yml) fails if the version of the extension (e.g. `version = "0.1.0"`) is the same as the version of a third-party dependency like `zed_extensions_api = "0.1.0"`.
2. The version bump script seems to increment all versions `0.1.0` found in `Cargo.toml`, instead of 
3. This causes the resulting Cargo.toml to contain an invalid version for `zed_extension_api = "0.1.1"` which does not exist: https://crates.io/crates/zed_extension_api/versions
4. This crashes the version bump script, thus likely preventing the release of a new extension version.

Here's the related log output causing the crash. Maybe this can be reproduced locally when running the GitHub action for the specific version `zed-extensions/harper@0.1.0` and trying to bump to `0.1.1`? https://github.com/zed-extensions/harper/actions/runs/20708549089/job/59443801797#step:6:50

### Current vs. Expected behavior

Current behaviour: The Zed-provided [`bump_extension_version` GitHub action](https://github.com/zed-industries/zed/blob/main/.github/workflows/extension_bump.yml) crashes and no new extension version is released. See details above.

Expected behaviour: The extension version should be possible to bump from `0.1.0` to `0.1.1` while not modifying any versions in the `[dependencies]` section, thus preserving `zed_extensions_api = "0.1.0"`.

### Zed version and system specs

N/A - this is caused by the Zed extensions GitHub action defined in this repository, but not directly related to the Zed editor itself.

### Attach Zed log file

<details><summary>Zed.log</summary>

```txt
+ OLD_VERSION=0.1.0
+ BUMP_FILES=("extension.toml")
+ [[ -f Cargo.toml ]]
+ BUMP_FILES+=("Cargo.toml")
+ bump2version --verbose --current-version 0.1.0 --no-configured-files patch extension.toml Cargo.toml
Could not read config file at .bumpversion.cfg
Attempting to increment part 'patch'
Values are now: major=0, minor=1, patch=1
New version will be '0.1.1'
Asserting files extension.toml, Cargo.toml contain the version string...
Would write to config file .bumpversion.cfg:
new_version=0.1.1
[bumpversion]
current_version = 0.1.1


Would prepare Git commit
Would add changes in file 'extension.toml' to Git
Would add changes in file 'Cargo.toml' to Git
Would commit to Git with message 'Bump version: 0.1.0 → 0.1.1'
Would tag 'v0.1.1' with message 'Bump version: 0.1.0 → 0.1.1' in Git and not signing
+ [[ -f Cargo.toml ]]
+ cargo update --workspace
    Updating crates.io index
error: failed to select a version for the requirement `zed_extension_api = "^0.1.1"`
candidate versions found which didn't match: 0.7.0, 0.6.0, 0.5.0, ...
location searched: crates.io index
required by package `harper_zed v0.1.1 (/home/runner/work/harper/harper)`
Error: Process completed with exit code 101.
```

</details>


### Relevant Zed settings

<details><summary>settings.json</summary>

<!-- Paste your settings inside the code block. -->
```json

```

</details>


### Relevant Keymap

<details><summary>keymap.json</summary>

<!-- Paste your keymap file inside the code block. -->
```json

```

</details>


### (for AI issues) Model provider details

_No response_

### If you are using WSL on Windows, what flavor of Linux are you using?

None

--- a/.github/workflows/extension_bump.yml
+++ b/.github/workflows/extension_bump.yml
@@ -91,7 +91,11 @@ jobs:
             BUMP_FILES+=("Cargo.toml")
         fi
 
-        bump2version --verbose --current-version "$OLD_VERSION" --no-configured-files ${{ inputs.bump-type }} "${BUMP_FILES[@]}"
+        bump2version \
+            --search "version = \"{current_version}"\" \
+            --replace "version = \"{new_version}"\" \
+            --current-version "$OLD_VERSION" \
+            --no-configured-files ${{ inputs.bump-type }} "${BUMP_FILES[@]}"
 
         if [[ -f "Cargo.toml" ]]; then
             cargo update --workspace

--- a/tooling/xtask/src/tasks/workflows/extension_bump.rs
+++ b/tooling/xtask/src/tasks/workflows/extension_bump.rs
@@ -261,7 +261,11 @@ fn bump_version(current_version: &JobOutput, bump_type: &WorkflowInput) -> (Step
                 BUMP_FILES+=("Cargo.toml")
             fi
 
-            bump2version --verbose --current-version "$OLD_VERSION" --no-configured-files {} "${{BUMP_FILES[@]}}"
+            bump2version \
+                --search "version = \"{{current_version}}"\" \
+                --replace "version = \"{{new_version}}"\" \
+                --current-version "$OLD_VERSION" \
+                --no-configured-files {} "${{BUMP_FILES[@]}}"
 
             if [[ -f "Cargo.toml" ]]; then
                 cargo update --workspace

